### PR TITLE
Add support for resource record sets

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: docs/dns-usage.rst [hunk 1]
```diff
@@ -104,23 +104,23 @@ Each managed zone exposes a read-only set of resource records:
      >>> from gcloud import dns
      >>> client = dns.Client(project='PROJECT_ID')
      >>> zone = client.zone('acme-co')
-     >>> records, page_token = zone.list_resources()  # API request
+     >>> records, page_token = zone.list_resource_record_sets()  # API request
```
Comment (dhermes) [inline docs/dns-usage.rst]: This rename was unexpected until you implemented?

File: gcloud/dns/zone.py [hunk 2]
```diff
@@ -267,3 +268,47 @@ def delete(self, client=None):
...
+        if page_token is not None:
+            params['pageToken'] = page_token
+
+        path = '/projects/%s/managedZones/%s/rrsets' % (
+            self.project, self.name)
+        client = self._require_client(client)
+        conn = client.connection
+        resp = conn.api_request(method='GET', path=path, query_params=params)
+        zones = [ResourceRecordSet.from_api_repr(resource, self)
+                 for resource in resp['rrsets']]
```
Comment (dhermes) [inline gcloud/dns/zone.py]: `resp.get('rrsets', ())`?

File: gcloud/dns/resource_record_set.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        :type zone: :class:`gcloud.dns.zone.ManagedZone`
+        :param zone: A zone which holds one or more record sets.
+
+        :rtype: :class:`gcloud.dns.zone.ResourceRecordSet`
+        :returns: RRS parsed from ``resource``.
+        """
+        name = resource['name']
+        record_type = resource['type']
+        ttl = int(resource['ttl'])
+        rrdatas = resource['rrdatas']
```
Comment (dhermes) [inline gcloud/dns/resource_record_set.py]: How crucial are each of these? (i.e. will this end up with a flaky `KeyError` like we've seen before with pub/sub, and if we do see the `KeyError`, is it fatal or could the code go on without some values?)

File: docs/dns-usage.rst (hunk 1 above)
Comment (tseaver) [inline docs/dns-usage.rst]: Yes.  It didn't seem right to use such a vague / confusing name.

File: gcloud/dns/zone.py (hunk 2 above)
Comment (tseaver) [inline gcloud/dns/zone.py]: `rrsets` is a hard requirement for the API (its only point, actually).  I wouldn't harden it unless we discover the backend is cheating.

File: gcloud/dns/resource_record_set.py (hunk 3 above)
Comment (tseaver) [inline gcloud/dns/resource_record_set.py]: They are hard requirements:  you can't use the zone at all if any are missing.

**Comment by dhermes** [comment]
LGTM

You know I'm always on the fence about if `KeyError` is sufficient enough explanation to the user of why the backend is lying to us and them, but I'm on the LGTM side of the fence today, proceed.

**Pull request merged**